LegoDimensions/LegoPortal.cs

```diff
@@ -587,9 +587,11 @@ private void ReadThread(object? obj)
                             else if (message.MessageCommand == MessageCommand.None && cmdId != null && cmdId.MessageCommand == MessageCommand.TagList)
                             {
                                 _presentTags.Clear();
+                                byte[] uid = new byte[7];
                                 for (int i = 0; i < message.Payload.Length / 2; i++)
                                 {
-                                    PresentTag presentTag = new PresentTag((Pad)(message.Payload[i * 2] >> 4), (TagType)message.Payload[i * 2 + 1], (byte)(message.Payload[i * 2] & 0xF));
+                                    Array.Copy(message.Payload, i * 2 + 4, uid, 0, uid.Length);
+                                    PresentTag presentTag = new PresentTag((Pad)(message.Payload[i * 2] >> 4), (TagType)message.Payload[i * 2 + 1], (byte)(message.Payload[i * 2] & 0xF), uid);
```

Comment:
I don't think you are receiving the UID from this command. See here: https://github.com/Ellerbach/LegoDimensions/blob/main/LegoDimensionsProtocol.md#list-tags-0xd0

It means that in this case, you have to get it another way from the UID stored in _padTag by getting it from the ID.